M2.el

```diff
@@ -306,11 +306,11 @@ can be executed with \\[M2-send-to-program]."
   interpreter using \\[comint-send-input]."
   (interactive)
   (cond ((save-excursion
-	   (search-backward-regexp "\\({\\*\\|^\\)")
-	   ;; example: {*FunctionBody[../../d/startup.m2.in:123:19-123:21]*}
-	   ;; example: {*Function[../../m2/res.m2:191:40-202:36]*}
+	   (search-backward-regexp "\\(-\\*\\|^\\)")
```

Comment:
I believe you may need to also add `--` to this regexp.

Reply:
It doesn't appear that `--Function[...]` ever will be output.  A quick `grep` shows that it's referenced in `LAYOUT`, so maybe this used to be the case?